react/src/SearchVariants/components/Autocomplete.tsx

```diff
@@ -1,20 +1,112 @@
-import React, { useState } from "react";
-import { CircularProgress, OutlinedInput, useTheme } from "@material-ui/core";
-import { Search } from "@material-ui/icons";
+import React, { useMemo, useState } from "react";
+import {
+    Button,
+    CircularProgress,
+    Divider,
+    InputAdornment,
+    makeStyles,
+    Menu,
+    MenuItem,
+    OutlinedInput,
+    useTheme,
+} from "@material-ui/core";
+import { ArrowDropDown, Check, Search } from "@material-ui/icons";
 import { Autocomplete } from "@material-ui/lab";
+import { snakeCaseToTitle } from "../../functions";
 import { useGenesQuery } from "../../hooks/genes";
 import { GeneAlias } from "../../typings";
 
+type SearchCategory = "gene" | "region" | "variant_position" | "rsld";
+const searchCategoryList: SearchCategory[] = ["gene", "region", "variant_position", "rsld"];
+
 interface GeneAutocompleteProps {
     fullWidth?: boolean;
     onSearch?: () => void;
     onSelect: (result: GeneAlias) => void;
 }
 
+interface SearchCategorySelectProps {
+    value: SearchCategory;
+    onSelect: (value: SearchCategory) => void;
+}
+
+const useSelectStyles = makeStyles(theme => ({
+    check: {
+        marginLeft: theme.spacing(2),
+    },
+    divider: {
+        margin: theme.spacing(0, 1),
+    },
+}));
+
+function SearchCategorySelect(props: SearchCategorySelectProps) {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+    const classes = useSelectStyles();
+
+    return (
+        <>
+            <Button onClick={event => setAnchorEl(event.currentTarget)}>
+                {snakeCaseToTitle(props.value)}
+                <ArrowDropDown />
+            </Button>
+            <Divider className={classes.divider} orientation="vertical" flexItem />
+            <Menu
+                anchorOrigin={{ horizontal: "left", vertical: "bottom" }}
+                transformOrigin={{ horizontal: "left", vertical: "top" }}
+                anchorEl={anchorEl}
+                keepMounted
+                open={!!anchorEl}
+                onClose={() => setAnchorEl(null)}
+                variant="menu"
```

Comment:
Was getting a cryptic console warning about these settings, evidently you need to manually set                 `getContentAnchorEl={null}` to pacify it.

Reply:
Oh nice, this change actually doubles as a way to make the popover appear below the Autocomplete bar so that you can actually see the changes made.